src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -651,7 +651,7 @@ impl<O: ForestObligation> ObligationForest<O> {
         }
 
         node_rewrites.truncate(0);
-        self.node_rewrites.replace(node_rewrites);
+        std::mem::replace(&mut self.node_rewrites, node_rewrites);
```

Comment:
Nit: Pre-existing, but why not just `self.node_writes = node_rewrites;`?

Reply:
Good point. I will update before landing.